### PR TITLE
Api

### PR DESCRIPTION
**Pull request by Itswali**
### IN THIS PR

- Use API to Store the book data.
- Use APi to load the data.
- Refactor the code for ApI.
- Add function for adding and removing a book

### CODE REVIEW

File: src/redux/fetchApi.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
```
Comment (TedLivist) [inline src/redux/fetchApi.js:2]: - [ ] You are importing `axios` here but you did not include it as a package in the `package.json` file so I get an error when trying to start your app.

  You can fix this by installing `axios`:

  ```
  npm install axios
  ```

  image

File: src/redux/fetchApi.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+export const addBookAPI = createAsyncThunk(ADD_BOOK_API, async (payload, thunkAPI) => {
+  await axios.post(URL, payload);
+  return thunkAPI.dispatch(getAllBooks());
+});
+
+export const removeBookAPI = createAsyncThunk(REMOVE_BOOK_API, async (payload, thunkAPI) => {
+  await axios.delete(`${URL}/${payload.id}`);
+  return thunkAPI.dispatch(getAllBooks());
+});
+
```
Comment (TedLivist) [inline src/redux/fetchApi.js]: - [ ] Your book delete implementation does not work. I am able to add books and the books are fetched from the API, but I cannot delete a book from the API. Kindly look into that and resolve it

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Itswali,

Good job so far!
✔️ Books can be added to API
✔️ Books are fetched
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/fetchApi.js (hunk 1 above)
Comment (Itswali) [reply src/redux/fetchApi.js:2]: I reinstalled the axios. I don't seem to know why it wasnot working it was working fine on my side.

File: src/redux/fetchApi.js (hunk 2 above)
Comment (Itswali) [reply src/redux/fetchApi.js]: I fixed the functionality of removeBook. ThankYou for noticing

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


## Optional suggestions


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- NA
### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**